airsonic-main/src/test/java/org/airsonic/player/service/search/SearchServiceSpecialGenreTestCase.java

```diff
@@ -0,0 +1,242 @@
+import org.airsonic.player.service.SearchService;
+import org.airsonic.player.service.SettingsService;
+import org.airsonic.player.util.HomeRule;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+@ContextConfiguration(
+        locations = {
+                "/applicationContext-service.xml",
+                "/applicationContext-cache.xml",
+                "/applicationContext-testdb.xml",
+                "/applicationContext-mockSonos.xml" })
+@DirtiesContext(
+        classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+/*
+ * Tests to clarify strings that can be used in genre.
+ */
+public class SearchServiceSpecialGenreTestCase {
+
+    @ClassRule
+    public static final SpringClassRule classRule = new SpringClassRule() {
+        HomeRule homeRule = new HomeRule();
+
+        @Override
+        public Statement apply(Statement base, Description description) {
+            Statement spring = super.apply(base, description);
+            return homeRule.apply(spring, description);
+        }
+    };
+
+    @Rule
+    public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+    @Autowired
+    private MediaScannerService mediaScannerService;
+
+    @Autowired
+    private MusicFolderDao musicFolderDao;
+
+    @Autowired
+    private DaoHelper daoHelper;
+
+    @Autowired
+    private SearchService searchService;
+
+    @Autowired
+    private SettingsService settingsService;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Autowired
+    ResourceLoader resourceLoader;
+
+    @Before
+    public void setup() throws Exception {
+        populateDatabase();
+    }
+
+    private static boolean dataBasePopulated;
+
+    private static Function<String, String> resolvePath = (childPath) ->{
+        return MusicFolderTestData.resolveBaseMediaPath() + childPath;
+    };
+    
+    private List<MusicFolder> musicFolders;
+    
+    private List<MusicFolder> getTestMusicFolders() {
+        if (isEmpty(musicFolders)) {
+            musicFolders = new ArrayList<>();
+
+            File musicDir = new File(resolvePath.apply("Search/SpecialGenre"));
+            musicFolders.add(new MusicFolder(1, musicDir, "accessible", true, new Date()));
+
+        }
+        return musicFolders;
+    }
+
+    private int count = 1;
```

Comment:
Can't this variable be in `populateDatabase` ?

Reply:
This is the count called from ``@Before``, so it is written there.

I wrote it with reference to other tests, but it seems that Scan may not be performed correctly if it is called almost simultaneously from multiple methods.
(A painful workaround ...)

Reply:
Please add a small comment explaining this then :)

Reply:
I revised that according to your advice.